File: mpp-proiect-repository-cristinahendre/lab7(spring)/lab6/modul/DomainModule/src/main/java/Domain/Validators/VoluntarValidator.java
package Domain.Validators;

import Domain.Voluntar;

public class VoluntarValidator implements  Validator<Voluntar> {

    /**
     * Valideaza un voluntar
     * @param entity ->voluntarul
     * @throws ValidationException
     */
    @Override
    public void validate(Voluntar entity) throws ValidationException {
        if(entity.getEmail().equals("") || entity.getEmail().equals(" "))
            throw new ValidationException("Email-ul este vid.");

        if(entity.getParola().equals("") || entity.getParola().equals(" "))
            throw  new ValidationException("Parola este vida.");


    }
}

File: mpp-proiect-repository-cristinahendre/lab10/CSharp/Lab5/Domain/Validator/VoluntarValidator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Cerinta1.Domain.Validator
{
    class VoluntarValidator : IValidator<Voluntar>
    {
     /*
     * Valideaza un voluntar
     * @param entity ->voluntarul
     * @throws ValidationException
     */
        public void Validate(Voluntar entity)
        {
            if (entity.Email.Equals("") || entity.Email.Equals(" "))
                throw new ValidationException("Email vid.");
            if (entity.Parola.Equals("") || entity.Parola.Equals(" "))
                throw new ValidationException("Parola e vida.");
        }
    }
}
